HCCGo/app/js/jobViewCtrl.js

```diff
@@ -117,10 +117,13 @@ jobViewModule.controller('jobViewCtrl', ['$scope', '$log', '$timeout', 'connecti
       // If the user clicks cancel, localFile will be undefined
       if (!localFile) return;
 
-      if (!downloadRemoteFile($scope.job[fileType], localFile)) {
-        fs.writeFile(localFile, $scope.job[fileType], function(err) {});
-      }
-      notifierService.success(path.basename(localFile).trim() + " Was Successfully Saved!", "Save Successful!");
+      downloadRemoteFile($scope.job[fileType], localFile).then(function(flag) {
+        if (flag) {
+          fs.writeFile(localFile, $scope.job[fileType], function(err) {});
+        }
+      });
+      
+      notifierService.success(path.basename(localFile).trim() + " was Successfully Saved!", "Save Successful!");
```

Comment:
Should probably move this success inside the `then`, to it triggers after the file has been written.